src/sources/hydra.rs

```diff
@@ -0,0 +1,284 @@
+        stage.output.send(json_evt.into()).await.or_panic()?;
+        stage.ops_count.inc(1);
+
+        // Apply CborTx events for any txs
+        match next.payload {
+            HydraMessagePayload::TxValid { tx } => {
+                let evt = ChainEvent::Apply(point.clone(), Record::CborTx(tx));
+                stage.output.send(evt.into()).await.or_panic()?;
+                stage.ops_count.inc(1);
+
+                stage.current_slot.set(point.slot_or_default() as i64);
+                stage.ops_count.inc(1);
+            }
+            _ => (),
+        };
+        Ok(())
+    }
+}
+
+fn intersect_from_config(intersect: &IntersectConfig) -> WorkerIntersect {
+    match intersect {
+        IntersectConfig::Origin => {
+            info!("starting from Origin");
+            WorkerIntersect::ProcessMessages
+        }
+        IntersectConfig::Tip => {
+            panic!("intersecting tip not currently supported with hydra as source")
+        }
+        IntersectConfig::Point(slot, hash_str) => {
+            info!("intersecting specific point");
+            let hash = hex::decode(hash_str).expect("valid hex hash");
+            WorkerIntersect::SkipUntil(slot.clone(), hash)
+        }
+        IntersectConfig::Breadcrumbs(_) => {
+            panic!("intersecting breadcrumbs not currently supported with hydra as source")
+        }
+    }
+}
+
+#[async_trait::async_trait(?Send)]
+impl gasket::framework::Worker<Stage> for Worker {
+    async fn bootstrap(stage: &Stage) -> Result<Self, WorkerError> {
+        debug!("connecting to hydra WebSocket");
+
+        let url = &stage.config.hydra_socket_url;
+        let (socket, _) = connect_async(url).await.expect("Can't connect");
+        let worker = Self {
+            socket,
+            intersect: intersect_from_config(&stage.intersect),
+        };
+
+        Ok(worker)
+    }
+
+    async fn schedule(&mut self, _stage: &mut Stage) -> Result<WorkSchedule<Message>, WorkerError> {
+        let next_msg = self.socket.next().await.transpose().or_restart()?;
+
+        Ok(match next_msg {
+            Some(message) => WorkSchedule::Unit(message),
+            None => WorkSchedule::Idle,
+        })
+    }
+
+    async fn execute(&mut self, message: &Message, stage: &mut Stage) -> Result<(), WorkerError> {
+        match message {
+            Message::Text(text) => {
+                debug!("Hydra message: {}", text);
+                match serde_json::from_str::<HydraMessage>(text) {
+                    Ok(hydra_message) => {
+                        self.process(stage, hydra_message).await
+                    }
+                    Err(err) => {
+                        error!("Failed to parse Hydra message: {}", err);
+                        Ok(())
+                    }
+                }
+            }
+            _ => Ok(()),
+        }
+    }
+}
+
+#[derive(Deserialize)]
+pub struct Config {
+    pub hydra_socket_url: String,
```

Comment:
Probably this is more correct:

```suggestion
    pub hydra_socket_path: String,
```